src/darwin/Framework/CHIP/CHIPOperationalCredentialsDelegate.mm

```diff
@@ -326,8 +326,8 @@ static BOOL isRunningTests(void)
         }
     }
     if (!haveRootCert) {
-        ReturnErrorOnFailure(rcac_dn.AddAttribute(chip::ASN1::kOID_AttributeType_ChipRootId, mIssuerId));
-        ReturnErrorOnFailure(rcac_dn.AddAttribute(chip::ASN1::kOID_AttributeType_ChipFabricId, fabricId));
+        ReturnErrorOnFailure(rcac_dn.AddAttribute_MatterRCACId(mIssuerId));
```

Comment:
So... This is fine as far as it goes, but the actual TODO was about seeing whether here was a way to make the "link up the NOC and ICA cert and RCA cert just right" code that's more or less duplicated across all the credential providers harder to mess up.

For example, maybe we should have a utility for "create an operational root cert" that takes the validity start, validity end, issuer id, fabric id, and key and does the setup of the DN, the listing of the DN twice in the X509CertRequestParams, etc.  And then returns some piece of data that we can pass to the "create an ICA cert" or "create a NOC" bits that will automatically set the right "parent" DN, etc....

Reply:
I think this comment can be reviewed and addressed separately from this PR. I added this comment to #14314

Please to merge this PR as is.